Terminal.Gui/Views/PanelView.cs

```diff
@@ -246,5 +302,31 @@ public override void Redraw (Rect bounds)
 			ClearLayoutNeeded ();
 			ClearNeedsDisplay ();
 		}
+
+		/// <summary>
+		/// Virtual method to invoke the <see cref="Clicked"/> event.
+		/// </summary>
+		public virtual void OnClicked ()
+		{
+			if (UseEffect3DAnimation) {
+				Border.Effect3D = false;
+				var x = X;
+				if (Border.Effect3DOffset.X > 0) {
+					X += 1;
```

Comment:
Just want to echo @tig comment here that we don't want to physically move the Button here.  What if there is another View with `Pos.Right(myButton) + 1` it would judder with the button wouldn't it?  

I think we want a render effect not a position effect.

Reply:
The `Mainloop.Invoke` will not be used now. It will be moved down on button pressed, moved up on button released and only after the animation the action is executed and will not interfere with the what the action does.

Reply:
> I think we want a render effect not a position effect.

Without a position effect there isn't a true animation. I tried by only clear the shadow but the effect it isn't so good. 

> What if there is another View with `Pos.Right(myButton) + 1` it would judder with the button wouldn't it?

It will only be side by side. But I recommend using more space between the controls, of course.

Reply:
I agree that it should 'appear to move' just that it's `X` should not change as that could cause problems for other controls' layouts (e.g. if they use relative positioning).

Reply:
Ok I decided not moving the button but only hide and show the shadow. So, this will avoid issue with relative layout positions used by others controls.